src/main/scala/fm/http/LinkedHttpContentReader.scala

```diff
@@ -39,6 +39,20 @@ final class LinkedHttpContentReader(is100ContinueExpected: Boolean, head: Future
   private[this] val foldLeftCalled: AtomicBoolean = new AtomicBoolean(false)
   @volatile private[this] var current: Future[Option[LinkedHttpContent]] = head
 
+  // Access the LinkedHttpContent
+  def useLinkedHttpContent[T](f: LinkedHttpContent => Future[T]): Future[T] = {
```

Comment:
I'm not sure what you think this accomplishes...  All you've really done is create an accessor for the `current` variable.  There's a reason everything goes through the foldLeft method.